index.js

```diff
@@ -90,6 +98,57 @@ function inject (dispatchFunc, options, callback) {
   }
 }
 
+function Chain (dispatch, option = {}) {
+  this.option = {
+    ...option
+  }
+  this.dispatch = dispatch
+}
+
+const httpMethods = [
+  'delete',
+  'get',
+  'head',
+  'options',
+  'patch',
+  'post',
+  'put',
+  'trace'
+]
+
+httpMethods.forEach(method => {
+  Chain.prototype[method] = function (url) {
+    this.option.url = url
+    this.option.method = method.toUpperCase()
+    return this
+  }
+})
+
+const chainMethods = [
+  'body',
+  'headers',
+  'payload',
+  'query'
+]
+
+chainMethods.forEach(method => {
+  Chain.prototype[method] = function (value) {
+    this.option[method] = value
+    return this
+  }
+})
+
+Chain.prototype.end = function (callback) {
+  doInject(this.dispatch, this.option, callback)
+}
+
+Object.getOwnPropertyNames(Promise.prototype).forEach(method => {
+  if (method === 'constructor') return
+  Chain.prototype[method] = function (...args) {
+    return doInject(this.dispatch, this.option)[method](...args)
+  }
+})
```

Comment:
Provides backward compatibility for promise.
Tell me if there is a better way!

Reply:
I think this is ok. I would just have added a `then` method, but this is better.

The implementation has a side effect, as invoking multiple times `.then()` will issue multiple calls vs waiting for the same one. You'll have to get hold of a promise and the keep calling the methods on that one instead, i.e. the same `Chain` object wraps only one promise.

Reply:
Got it. But I'm wondered is there any cases should we write codes like that?

However, if different `then` returns different Promise, we can test different cases using one `Chain` Instance, just like:
```js
const chain = inject(dispatch)
chain.get('/').then()
chain.post('/').then()
```
I think the above case might be more common.

Reply:
I think we should throw in the above case, you can’t set the method twice